Drop-In/src/main/java/com/braintreepayments/api/BaseActivity.java

```diff
@@ -4,75 +4,46 @@
 import android.os.Bundle;
 import android.text.TextUtils;
 
-import com.braintreepayments.api.exceptions.InvalidArgumentException;
-import com.braintreepayments.api.models.Authorization;
-import com.braintreepayments.api.models.ClientToken;
-import com.braintreepayments.api.models.Configuration;
-import com.braintreepayments.api.models.PaymentMethodNonce;
-
 import org.json.JSONException;
 
 import androidx.annotation.Nullable;
 import androidx.appcompat.app.AppCompatActivity;
+import androidx.fragment.app.FragmentActivity;
 
 class BaseActivity extends AppCompatActivity {
 
     static final String EXTRA_CONFIGURATION_DATA = "com.braintreepayments.api.EXTRA_CONFIGURATION_DATA";
 
     protected DropInRequest mDropInRequest;
-    protected BraintreeFragment mBraintreeFragment;
-    protected Configuration mConfiguration;
     protected boolean mClientTokenPresent;
 
+    private DropInClient dropInClient;
+
     @Override
     protected void onCreate(@Nullable Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
 
-        if (savedInstanceState != null) {
-            try {
-                mConfiguration = Configuration.fromJson(savedInstanceState.getString(EXTRA_CONFIGURATION_DATA));
-            } catch (JSONException ignored) {}
-        }
-
-        mDropInRequest = getIntent().getParcelableExtra(DropInRequest.EXTRA_CHECKOUT_REQUEST);
+        mDropInRequest = getIntent().getParcelableExtra(DropInClient.EXTRA_CHECKOUT_REQUEST);
     }
 
     @Override
     protected void onSaveInstanceState(Bundle outState) {
         super.onSaveInstanceState(outState);
-        if (mConfiguration != null) {
-            outState.putString(EXTRA_CONFIGURATION_DATA, mConfiguration.toJson());
-        }
     }
 
-    protected BraintreeFragment getBraintreeFragment() throws InvalidArgumentException {
-        if (TextUtils.isEmpty(mDropInRequest.getAuthorization())) {
-            throw new InvalidArgumentException("A client token or tokenization key must be " +
-                    "specified in the " + DropInRequest.class.getSimpleName());
-        }
-
-        try {
-            mClientTokenPresent = Authorization.fromString(mDropInRequest.getAuthorization())
-                    instanceof ClientToken;
-        } catch (InvalidArgumentException e) {
-            mClientTokenPresent = false;
+    DropInClient getDropInClient() {
+        // lazily instantiate dropInClient for testing purposes
+        if (dropInClient != null) {
+            return dropInClient;
         }
```

Comment:
Why are we doing this instead of making DropInClient a property?

Reply:
We inject a `DropInClient` into `DropInActivity` by overriding this method in the unit tests. We want to find a better way to inject mocks, but here we borrow the existing `BraintreeFragment` mocking pattern.